tensorflow/python/ops/numpy_ops/np_array_ops.py

```diff
@@ -560,6 +561,23 @@ def _reduce(tf_fn,
       tf_fn(input_tensor=a.data, axis=axis, keepdims=keepdims))
 
 
+# TODO (DarrenZhang01): Add `axis` support to the `size` API.
+@np_utils.np_doc('size')
+def size(x, axis=None):   # pylint: disable=missing-docstring
+  x = asarray(x).data
+  if axis is not None:
+    raise NotImplementedError("axis argument is not supported in the current "
+                              "`np.size` implementation")
+  if isinstance(x, (int, float, np.int32, np.int64,
+                    np.float32, np.float64)):
+    return 1
+  elif isinstance(x, (np_arrays.ndarray, np.ndarray)) or tf.is_tensor(x):
+    return prod(tuple(x.shape))
```

Comment:
This still doesn't handle Tensors with partially defined shapes. 

x.shape returns a [TensorShape](https://www.tensorflow.org/api_docs/python/tf/TensorShape). If the shape [is_fully_defined](https://www.tensorflow.org/api_docs/python/tf/TensorShape#is_fully_defined), then the code as written works OK, but that is not guaranteed. If the shape isn't fully defined, you'll have to call the corresponding TensorFlow [size](https://www.tensorflow.org/api_docs/python/tf/size) function, and return that as a result.